lib/ansible/modules/network/cloudengine/ce_vxlan_tunnel.py

```diff
@@ -0,0 +1,936 @@
+            self.updates_cmd.append("interface %s" % nve_name)
+            self.updates_cmd.append("undo mode l3")
+            self.changed = True
+        else:
+            self.module.fail_json(
+                msg='Error: Can not configure undo mode l2.')
+
+    def config_delete_source_ip(self, nve_name, source_ip):
+        """nve source ip"""
+
+        if not self.is_nve_source_ip_exist(nve_name, source_ip):
+            return
+        ipaddr = "0.0.0.0"
+        cfg_xml = CE_NC_MERGE_NVE_SOURCE_IP_PROTOCOL % (
+            nve_name, ipaddr)
+        recv_xml = set_nc_config(self.module, cfg_xml)
+        self.check_response(recv_xml, "DELETE_SOURCE_IP")
+        self.updates_cmd.append("interface %s" % nve_name)
+        self.updates_cmd.append("undo source %s" % source_ip)
+        self.changed = True
+
+    def config_delete_vni_peer_ip(self, nve_name, vni_id, peer_ip_list):
+        """remove vni peer ip"""
+
+        for peer_ip in peer_ip_list:
+            if not self.is_vni_peer_list_exist(nve_name, vni_id, peer_ip):
+                self.module.fail_json(msg='Error: The %s is not exist' % peer_ip)
+        config = self.get_current_config(vni_id, peer_ip_list)
+        if not config:
+            cfg_xml = CE_NC_DELETE_VNI_PEER_ADDRESS_IP_HEAD % (
+                nve_name, vni_id)
+            for peer_ip in peer_ip_list:
+                cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_DELETE % peer_ip
+            cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_END
+        else:
+            cfg_xml = CE_NC_DELETE_PEER_ADDRESS_IP_HEAD % (
+                nve_name, vni_id)
+            for peer_ip in peer_ip_list:
+                cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_DELETE % peer_ip
+            cfg_xml += CE_NC_DELETE_PEER_ADDRESS_IP_END
+
+        recv_xml = set_nc_config(self.module, cfg_xml)
+        self.check_response(recv_xml, "DELETE_VNI_PEER_IP")
+        self.updates_cmd.append("interface %s" % nve_name)
+
+        for peer_ip in peer_ip_list:
+            cmd_output = "undo vni %s head-end peer-list %s" % (vni_id, peer_ip)
+            self.updates_cmd.append(cmd_output)
+
+        self.changed = True
+
+    def config_delete_vni_protocol_type(self, nve_name, vni_id, protocol_type):
+        """remove vni protocol type"""
+
+        if not self.is_vni_protocol_exist(nve_name, vni_id, protocol_type):
+            return
+
+        cfg_xml = CE_NC_DELETE_VNI_PROTOCOL % (nve_name, vni_id, protocol_type)
+        recv_xml = set_nc_config(self.module, cfg_xml)
+        self.check_response(recv_xml, "DELETE_VNI_PEER_PROTOCOL")
+        self.updates_cmd.append("interface %s" % nve_name)
+        self.updates_cmd.append(
+            "undo vni %s head-end peer-list protocol bgp " % vni_id)
+        self.changed = True
+
+    def check_params(self):
+        """Check all input params"""
+
+        # bridge_domain_id check
+        if self.bridge_domain_id:
+            if not self.bridge_domain_id.isdigit():
```

Comment:
specifying `type='int'` removes the need for this, so you only need to check the range.